[sailfish-browser] Change page background handling to follow page container.  Fixes JB#29662

rainemak opened the pull request:
Review @rojkov @mikkoharju 

Sharing via email and selecting contact ended up being tough nut to crack but now that works as well.

Also sharing via email and picking attachments is working and as well as "normal" facebook like sharing. This doesn't contain external dependencies.

Same approach will be automatically used for web prompts, password manager dialogs, etc.

Commit:
[sailfish-browser] Move page background handling to follow page container. Fixes JB#29662

rojkov commented on src/browser.qml:
```diff
@@ -57,4 +93,48 @@ ApplicationWindow {
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
+                        backgrounds = backgrounds.slice(index + 1)
```
could be just
backgrounds = backgrounds.slice(1)

rainemak replied:
True :)

rojkov commented on src/browser.qml:
```diff
@@ -57,4 +93,48 @@ ApplicationWindow {
+                    } else if (index > 0 && index < backgrounds.length - 1) {
+                        var firstPart = backgrounds.slice(0, index)
+                        var secondPart = backgrounds.slice(index + 1)
+                        backgrounds = firstPart.concat(secondPart)
```
splice?

rainemak replied:
Good point... Following should be enough.

`background.splice(index, 1)`

Will verify that original array updates correctly.

rainemak commented:
Rebased and updated.

rojkov commented on src/browser.qml:
```diff
@@ -48,4 +86,45 @@ ApplicationWindow {
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
```
I guess that given `index` is initialized to -1 these three conditions could be collapsed to only one

```
if (index >=  0 && backgrounds.length > 0) {
    background.splice(index, 1)
} else {
    backgrounds = []
}
```

rainemak replied:
That would work. Looks that when I wrote this a late night I wasn't thinking too much :)

Index is initialized to 0 but length > 0 guards first block. So, the else block takes care of the rest. Thanks.

Commit:
[sailfish-browser] Allow resize on sub pages

Background changed to be a square and virtual keyboard
observer of the overlay active only when overlay active (BrowserPage
active).

rainemak commented:
Added one commit on top that fixes content resizing on sub pages. @mikkoharju could give a look. What do you think about having a bit bigger background. As the background is not blended, I didn't see anything extra when peeking (popping) a page from deeper hierarchy.

E.g. Share => Facebook and pop

rainemak commented:
@antseppa could you check as well.

rojkov commented on src/browser.qml:
```diff
@@ -48,4 +86,41 @@ ApplicationWindow {
+            // Page and attached background is destroyed when the page gets destroyed.
+            onPageChanged: {
+                if (!page) {
+                    var index = 0
```
`index` should be better initialized to `-1`

rojkov replied:
or perhaps that would be too defensive style... LGTM

antseppa commented:
LGTM, works for me.
